src/components/sidebar-component/Sidebar.jsx

```diff
@@ -0,0 +1,71 @@
+                <input type="range" min="0" max="4" defaultValue={state.slider}  className="slider mt-16" onClick={(e)=>dispatch({type:"SET_SLIDER",payload:e.target.value})} />
+            </div>
+            <div className="category-container mt-32">
+                <h3>Categories</h3>
+                <div className="category mt-16 x-small-text">
+                    <input type="checkbox" id="action-figures" value="Action-figures" onClick={(e)=>dispatch({type:'CATEGORY_ACTION_FIGURES',payload:!state.categoryActionFigures})} checked={state.categoryActionFigures} />
+                <label htmlFor="male-figures">Action-figures</label>
+                </div>
+                <div className="category x-small-text">
+                    <input type="checkbox" id="t-shirts" name="T-shirts" onClick={(e)=>dispatch({type:'CATEGORY_T_SHIRTS',payload:!state.categoryTshirts})} checked={state.categoryTshirts}/>
+                <label htmlFor="female-figures">T-shirts</label>
+                </div>
+                <div className="category x-small-text">
+                    <input type="checkbox" id="masks" name="Masks" onClick={(e)=>dispatch({type:'CATEGORY_MASKS',payload:!state.categoryMasks})} checked={state.categoryMasks}/>
+                <label htmlFor="female-figures">Masks</label>
+                </div>
+                
+            </div>
+            <div className="ratings-container mt-32">
+                <h3>Ratings</h3>
+                <div className="rating mt-16 x-small-text">
+                    <input type="radio" id="stars_4" value="4" name="ratings" onClick={(e)=>dispatch({type:'FILTER_BY_RATING',payload:e.target.value})}/>
+                <label htmlFor="stars_4">4 stars and above</label>
+                </div>
+                <div className="rating x-small-text">
+                    <input type="radio" id="stars_3" name="ratings" value="3" onClick={(e)=>dispatch({type:'FILTER_BY_RATING',payload:e.target.value})} />
+                <label htmlFor="stars_3">3 stars and above</label>
+                </div>
+                <div className="rating x-small-text">
+                    <input type="radio" id="stars_2" name="ratings" value="2" onClick={(e)=>dispatch({type:'FILTER_BY_RATING',payload:e.target.value})} />
+                <label htmlFor="stars_2">2 stars and above</label>
+                </div>
+                <div className="rating x-small-text">
+                    <input type="radio" id="stars_1" name="ratings" value="1" onClick={(e)=>dispatch({type:'FILTER_BY_RATING',payload:e.target.value})} />
+                <label htmlFor="stars_1">1 star and above </label>
+                </div>
+            </div>
+
+            <div className="sort-filter-container mt-32">
+                <h3>Sort by</h3>
+                <div className="sort x-small-text">
+                    <input type="radio" id="low-to-high" name="sort_by" onClick={()=>dispatch({type:'PRICE_LOW_TO_HIGH',payload:'PRICE_LOW_TO_HIGH'})}/>
```

Comment:
for button's and input's click function define a function as onCLickHandler and put these in them it would look clean practise